File: app/src/main/java/com/ballastlane/android/texasam/app/ui/main/NavigationControllerActivityMain.java
package com.ballastlane.android.texasam.app.ui.main;

import com.ballastlane.android.baselibrary.common.base.BaseActivityInnerNavigation;
import com.ballastlane.android.baselibrary.common.base.BaseFragment;
import com.ballastlane.android.baselibrary.common.base.navigation.BaseActivityInnerNavigationController;
import com.ballastlane.android.texasam.R;
import com.ballastlane.android.texasam.app.ui.about.FrgAbout;
import com.ballastlane.android.texasam.app.ui.account.FrgAccount;
import com.ballastlane.android.texasam.app.ui.payment.FrgPayments;
import com.ballastlane.android.texasam.app.ui.sessions.FrgSessions;

import java.util.HashMap;


public class NavigationControllerActivityMain extends BaseActivityInnerNavigationController {

    private HashMap<String, BaseFragment> navFragments;

    public NavigationControllerActivityMain(BaseActivityInnerNavigation activity) {
        super(activity, R.id.container);
        initFragments();
    }

    private void initFragments() {
        this.navFragments = new HashMap<>();
        this.navFragments.put(getSection1Title(), FrgAbout.newInstance());
        this.navFragments.put(getSection2Title(), FrgSessions.newInstance());
        this.navFragments.put(getSection3Title(), FrgPayments.newInstance());
        this.navFragments.put(getSection4Title(), FrgAccount.newInstance());
    }

    private String getSection1Title() {
        return getContext().getString(R.string.nav_about);
    }

    private String getSection2Title() {
        return getContext().getString(R.string.nav_surveys);
    }


    private String getSection3Title() {
        return getContext().getString(R.string.nav_payments);
    }

    private String getSection4Title() {
        return getContext().getString(R.string.nav_account);
    }

    public void navigateToSection1() {
        navigateToRootLevel(this.navFragments.get(getSection1Title()), getSection1Title());
    }

    public void navigateToSection2() {
        navigateToRootLevel(this.navFragments.get(getSection2Title()), getSection2Title());
    }

    public void navigateToSection3() {
        navigateToRootLevel(this.navFragments.get(getSection3Title()), getSection3Title());
    }

    public void navigateToSection4() {
        navigateToRootLevel(this.navFragments.get(getSection4Title()), getSection4Title());
    }

}

File: sample/src/main/java/com/ballastlane/ballastlanebaseapp/app/ui/main/NavigationControllerActivityMain.kt
package com.ballastlane.ballastlanebaseapp.app.ui.main

import com.kogimobile.baselibrary.sample.R
import java.util.HashMap

/**
 * Created by Mariangela Salcedo (mariangelasalcedo@ballastlane.com) on 4/2/18.
 * Copyright (c) 2018 Ballast Lane Applications LLC. All rights reserved.
 */

class NavigationControllerActivityMain(activity: BaseActivityInnerNavigation) : BaseActivityInnerNavigationController(activity, R.id.container) {

    private var navFragments: HashMap<String, BaseFragment>? = null

    val section1Title: String
        get() = getContext().getString(R.string.nav_drawer_item_section_1)

    val section2Title: String
        get() = getContext().getString(R.string.nav_drawer_item_section_2)

    val section3Title: String
        get() = getContext().getString(R.string.nav_drawer_item_section_3)

    val section4Title: String
        get() = getContext().getString(R.string.nav_drawer_item_section_4)

    val subSection1Title: String
        get() = getContext().getString(R.string.nav_drawer_item_sub_section_1)

    init {
        initFragments()
    }

    private fun initFragments() {
        this.navFragments = HashMap<String, BaseFragment>()
        this.navFragments!![section1Title] = FrgNavigation.newInstance(getFragmentManager().getBackStackEntryCount())
        this.navFragments!![section2Title] = FrgEvents.newInstance()
        this.navFragments!![section3Title] = FrgUtils.newInstance()
        this.navFragments!![section4Title] = FrgRecyclerView.newInstance()
        this.navFragments!![subSection1Title] = TestFragment.newInstance()
    }

    fun navigateToSection1() {
        navigateToRootLevel(this.navFragments!![section1Title], section1Title)
    }

    fun navigateToSection1LowLevel(frg: BaseFragment, title: String) {
        navigateToLowLevel(frg, title)
    }

    fun navigateToSection2() {
        navigateToRootLevel(this.navFragments!![section2Title], section2Title)
    }

    fun navigateToSection3() {
        navigateToRootLevel(this.navFragments!![section3Title], section3Title)
    }

    fun navigateToSection4() {
        navigateToRootLevel(this.navFragments!![section4Title], section4Title)
    }

    fun navigateToSubSection1() {
        navigateToLowLevel(this.navFragments!![subSection1Title], subSection1Title)
    }
}
